geode-core/src/main/java/org/apache/geode/internal/cache/versions/RegionVersionVector.java

```diff
@@ -574,17 +609,20 @@ public void initializeVersionHolder(T mbr, RegionVersionHolder<T> otherHolder) {
     }
   }
 
-  private void updateLocalVersion(long version) {
-    boolean repeat = false;
+  void updateLocalVersion(long newVersion) {
+    boolean needToUpdate;
```

Comment:
I don't like the name "needToUpdate". I think a better name would be "needToTrySetAgain". It is only set to true when the compareAndSet fails and we need to retry. If it is set to true it does not mean we "needToUpdate" we just need to check again.